plugins/chat/assets/javascripts/discourse/routes/chat-index.js

```diff
@@ -5,6 +5,21 @@ export default class ChatIndexRoute extends DiscourseRoute {
   @service chat;
   @service chatChannelsManager;
   @service router;
+  @service siteSettings;
+  @service currentUser;
+
+  get hasThreads() {
+    if (!this.siteSettings.chat_threads_enabled) {
+      return false;
+    }
+    return this.currentUser?.chat_channels?.public_channels?.some(
```

Comment:
We also use this same logic in the `channel-list-public` and the `chat-footer` components now, maybe we can just have it in one place and import it?

Reply:
yes I saw it, I think that's an OK duplication for now

Reply:
No worries, we can do a cleanup of these duplications later.. not super urgent.